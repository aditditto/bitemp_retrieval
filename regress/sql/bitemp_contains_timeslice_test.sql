DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;

-- Contains
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz,
    '2020-10-15'::timestamptz
);

-- Outside effective range
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-25'::timestamptz,
    '2020-10-15'::timestamptz
);

-- Outside asserted range
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz,
    '2020-10-25'::timestamptz
);

SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2021-10-25'::timestamptz,
    '2021-10-25'::timestamptz
);

-- Contains
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity'),
    temporal_relationships.timeperiod('-infinity', 'infinity')
);

-- Outside effective range
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz),
    temporal_relationships.timeperiod('-infinity', 'infinity')
);

-- Outside asserted range
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity'),
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz)
);

-- Outside effective & asserted
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz),
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz)
);

SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz
);

-- Outside range
SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-25'::timestamptz
);

-- Outside range
SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-20'::timestamptz
);

SELECT interval_contains_now(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);

-- Outside range
SELECT interval_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity')
);
