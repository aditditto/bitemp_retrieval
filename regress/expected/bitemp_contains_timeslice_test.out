DROP EXTENSION bitemp_retrieval;
CREATE EXTENSION bitemp_retrieval;
-- Contains
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz,
    '2020-10-15'::timestamptz
);
 bitemp_contains_ts 
--------------------
 t
(1 row)

-- Outside effective range
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-25'::timestamptz,
    '2020-10-15'::timestamptz
);
 bitemp_contains_ts 
--------------------
 f
(1 row)

-- Outside asserted range
SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz,
    '2020-10-25'::timestamptz
);
 bitemp_contains_ts 
--------------------
 f
(1 row)

SELECT bitemp_contains_ts(
    temporal_relationships.timeperiod('2010-10-10'::timestamptz, '2020-10-20'::timestamptz),
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2021-10-25'::timestamptz,
    '2021-10-25'::timestamptz
);
 bitemp_contains_ts 
--------------------
 f
(1 row)

-- Contains
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity'),
    temporal_relationships.timeperiod('-infinity', 'infinity')
);
 bitemp_contains_now 
---------------------
 t
(1 row)

-- Outside effective range
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz),
    temporal_relationships.timeperiod('-infinity', 'infinity')
);
 bitemp_contains_now 
---------------------
 f
(1 row)

-- Outside asserted range
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity'),
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz)
);
 bitemp_contains_now 
---------------------
 f
(1 row)

-- Outside effective & asserted
SELECT bitemp_contains_now(
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz),
    temporal_relationships.timeperiod('-infinity', '1850-10-20'::timestamptz)
);
 bitemp_contains_now 
---------------------
 f
(1 row)

SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-15'::timestamptz
);
 interval_contains_ts 
----------------------
 t
(1 row)

-- Outside range
SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-25'::timestamptz
);
 interval_contains_ts 
----------------------
 f
(1 row)

-- Outside range
SELECT interval_contains_ts(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz),
    '2020-10-20'::timestamptz
);
 interval_contains_ts 
----------------------
 f
(1 row)

SELECT interval_contains_now(
    temporal_relationships.timeperiod('2020-10-10'::timestamptz, '2020-10-20'::timestamptz)
);
 interval_contains_now 
-----------------------
 f
(1 row)

-- Outside range
SELECT interval_contains_now(
    temporal_relationships.timeperiod('-infinity', 'infinity')
);
 interval_contains_now 
-----------------------
 t
(1 row)

